File: scripts/script/选择技能.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 1,["7"] = 1,["8"] = 5,["9"] = 5,["10"] = 5,["12"] = 8,["13"] = 10,["14"] = 8,["15"] = 6,["16"] = 16,["17"] = 17,["18"] = 18,["19"] = 19,["21"] = 22,["22"] = 22,["23"] = 22,["24"] = 22,["25"] = 23,["26"] = 22,["27"] = 22,["28"] = 22,["30"] = 16,["31"] = 32,["32"] = 33,["33"] = 32,["34"] = 36,["35"] = 36});
local ____exports = {}
local ____DialogUtil = require("solar.solar-common.util.game.DialogUtil")
local DialogUtil = ____DialogUtil.default
____exports.default = __TS__Class()
local _____9009_62E9_6280_80FD = ____exports.default
_____9009_62E9_6280_80FD.name = "选择技能"
function _____9009_62E9_6280_80FD.prototype.____constructor(self)
    se:onHeroLevelUp(function(unitEvent)
        self:SkillPanel(unitEvent)
    end)
end
function _____9009_62E9_6280_80FD.prototype.SkillPanel(self, unitEvent)
    local buttonTexts = {"test1", "test2"}
    local skillId = 0
    if isAuto then
    else
        DialogUtil:show(
            unitEvent.trigUnitOwnerId,
            "选择技能",
            function(____, i, text)
                self:LearnSkill(unitEvent.trigUnit, skillId)
            end,
            table.unpack(buttonTexts)
        )
    end
end
function _____9009_62E9_6280_80FD.prototype.LearnSkill(self, unit, skillId)
    SelectHeroSkill(unit, skillId)
end
function _____9009_62E9_6280_80FD.prototype.AutoSelectSkill(self)
end
return ____exports

File: src/script/选择技能.ts
import DialogUtil from "@/DialogUtil";
import UnitEvent from "@/event/UnitEvent";
import Random from "@/Random";
import RandomUtil from "@/RandomUtil";
import { d_显示cd的被动技能 } from "xlsx/技能/显示cd的被动技能";

export default class 选择技能{
    /**
     * 单位和已经学习的技能列表
     */
    private tmUnitSkill:Map<number,Map<number,string>> = new Map()
    constructor(){
        //点击加点按钮，弹出技能选择列表，列表项根据当前角色的技能进行0筛选
        se.onHeroLevelUp((unitEvent)=>{
           
            this.SkillPanel( unitEvent)
        })
    }
    /**
     * 技能选择面板
     */
    SkillPanel(unitEvent:UnitEvent){
        let buttonTexts = ["ab01"]
        let skillId = 0
        if(isAuto){

        }else{
            let playerId = GetPlayerId(GetOwningPlayer(unitEvent.trigUnit))
            DialogUtil.show(playerId, "选择技能", (i, text) => {
                this.LearnSkill(unitEvent.trigUnit,buttonTexts[i])
            }, ...buttonTexts)
        }
    }
    //获取可学习的技能列表
    CanLearnSkillList(unit:unit){
        let playerId = GetPlayerId(GetOwningPlayer(unit))
        //判断是否已经学习了这个技能
        let tmSkill = this.tmUnitSkill.get(playerId)
        if(!tmSkill){
            tmSkill = new Map()
            this.tmUnitSkill.set(playerId,tmSkill)
        }
        let tlSkill:string[] = []
        //筛选可学习的技能，并随机
        d_显示cd的被动技能.forEach((item)=>{
            //添加规则
            tlSkill.push(item.key)
        })

        let tl = RandomUtil.getRandomElementByObjArrays(4,"key",tlSkill)
        return tl
    }

    /**
     * 学习技能，参考技能学习到qwer配置
     */
    LearnSkill(unit:unit,skillId:string){
        let playerId = GetPlayerId(GetOwningPlayer(unit))
        //判断是否已经学习了这个技能
        let tmSkill = this.tmUnitSkill.get(playerId)
        if(!tmSkill){
            tmSkill = new Map()
            this.tmUnitSkill.set(playerId,tmSkill)
        }
        print(unit,skillId)
        
        UnitAddAbility(unit,skillId)
    }

    AutoSelectSkill(){
        
    }
}